Source/SimConnect/Source/SimConnectClient.cpp

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) 2024 Steven Frost and Orion Lyau
+
+#include "SimConnectClient.h"
+
+#include <SimConnect.h>
+
+// -----------------------------------------------------------------------------
+
+namespace SimConnect
+{
+
+// -----------------------------------------------------------------------------
+
+SimConnectClient::SimConnectClient( const std::string& ApplicationName )
+	: ApplicationName( ApplicationName )
+	, SimConnectHandle( INVALID_HANDLE_VALUE )
+{
+}
+
+// -----------------------------------------------------------------------------
+
+SimConnectClient::~SimConnectClient()
+{
+}
+
+// -----------------------------------------------------------------------------
+
+bool SimConnectClient::Initialize()
+{
+	if ( !OpenConnection() )
+	{
+		return false;
+	}
+
+	return true;
+}
+
+// -----------------------------------------------------------------------------
+
+bool SimConnectClient::Uninitialize()
+{
+	if ( !CloseConnection() )
+	{
+		return false;
+	}
+
+	return true;
+}
+
+// -----------------------------------------------------------------------------
+
+bool SimConnectClient::IsConnected() const
+{
+	return SimConnectHandle != INVALID_HANDLE_VALUE;
```

Comment:
Just a note: if we receive `SIMCONNECT_RECV_ID_QUIT`, it indicates that the sim has shut down -- might be worth checking for that in the future (since the SimConnect client will no longer be connected / in a valid state at that point).

Reply:
Thanks! Will include that in the next PR.